main.nf

```diff
@@ -318,11 +318,11 @@ if (!params.references_parsed && !params.skip_mirdeep){
     HAIRPIN="$hairpin"
     if [ \${MATURE: -3} == ".gz" ]; then
         gunzip -f \$MATURE
-        MATURE=\${MATURE: -3}
+        MATURE=\${MATURE%%.gz}
     fi
     if [ \${HAIRPIN: -3} == ".gz" ]; then
         gunzip -f \$HAIRPIN
-        HAIRPIN=\${HAIRPIN: -3}
+        HAIRPIN=\${HAIRPIN%%.gz}
     fi
 
     # Remove any special base characters from reference genome FASTA file
```

Comment:
the line below should be:
`sed  '/^[^>]/s/[^ATGCatgc]/N/g' $refgenome > genome.fa`

Reply:
and update below to `bowtie-build genome.fa genome`